Lib/test/test_typing.py

```diff
@@ -1482,6 +1482,15 @@ def test_cannot_instantiate(self):
         with self.assertRaises(TypeError):
             type(u)()
 
+    def test_cannot_iterate(self):
+        with self.assertRaises(TypeError):
+            iter(Union)
+        with self.assertRaises(TypeError):
+            list(Union)
```

Comment:
Also add tests for Any, List[T], Union[str, T], Tuple[T], Callable[..., T], Callable[[T], str], Annotated[T, ''], and all other with special subclass of _BaseGenericAlias.

Reply:
Done - except for `List[T]` and `Tuple[T]`, which need to remain iterable in order for unpacking to work.